bullet-featherstone/src/JointFeatures.cc

```diff
@@ -80,16 +80,42 @@ double JointFeatures::GetJointAcceleration(
 double JointFeatures::GetJointForce(
     const Identity &_id, const std::size_t _dof) const
 {
+  double results = gz::math::NAN_D;
   const auto *joint = this->ReferenceInterface<JointInfo>(_id);
   const auto *identifier = std::get_if<InternalJoint>(&joint->identifier);
-  if (identifier)
-  {
+
+  if (identifier) {
     const auto *model = this->ReferenceInterface<ModelInfo>(joint->model);
-    return model->body->getJointTorqueMultiDof(
-          identifier->indexInBtModel)[_dof];
+    auto feedback = model->body->getLink(identifier->indexInBtModel).m_jointFeedback;
+    const auto &link = model->body->getLink(identifier->indexInBtModel);
+    results = 0.0;
+    if (link.m_jointType == btMultibodyLink::eRevolute) {
+      // According to the documentation in btMultibodyLink.h, m_axesTop[0] is the
+      // joint axis for revolute joints.
+      Eigen::Vector3d axis = convert(link.getAxisTop(0));
+      math::Vector3 axis_converted(axis[0], axis[1], axis[2]);
+      btVector3 angular = feedback->m_reactionForces.getAngular();
+      math::Vector3<double> angularTorque(
+        angular.getX(),
+        angular.getY(),
+        angular.getZ());
+      // BUG: The total force is 2 times the cmd one see:
+      // https://github.com/bulletphysics/bullet3/discussions/3713
+      results += axis_converted.Dot(angularTorque);
+      return results / 2.0;
```

Comment:
Is the `results / 2.0` done to workaround the bug? Note that it was recently fixed upstream: https://github.com/bulletphysics/bullet3/pull/4583. I think we also found that total force is not always 2 times than it should be (based on example in https://github.com/gazebosim/gz-physics/issues/565 and comment: https://github.com/bulletphysics/bullet3/discussions/3713#discussioncomment-657881). 

Anyways, best to guard this with a bullet version check e.g. for users using latest bullet version built from source.

```
#if BT_BULLET_VERSION < 326
  // not sure if this is always true
  return results / 2.0; 
#else
  return results;
#endif
```

Reply:
Yes, the `results / 2.0` part is used as an ugly workaround in order to retrieve a data that is reasonable. I didn't know that the PR was merged when I have write the code.
Probably another workaround (for bullet version < 3.26) could be `measured_torque = measured_torque - applied_torque` ? But I don't know how to retrieve the information about the applied torque from the  `jointInfo` class.

Reply:
I just did some quick tests and saw that bullet clears the applied joint forces and torques after each step so I think we're not able to retrieve the applied torque at this point (e.g. `model->body->getJointTorqueMultiDof` just returns 0). We could cache the applied torque in `JointInfo` struct but we would need to clear them before every step. For now, I think we can just add a comment about this bug affecting bullet versions < 326 - basically what you had before, a link to https://github.com/bulletphysics/bullet3/discussions/3713 or https://github.com/gazebosim/gz-physics/issues/565.